rust/lance/src/index/cache.rs

```diff
@@ -45,6 +65,11 @@ impl IndexCache {
     }
 
     pub(crate) fn get_vector(&self, key: &str) -> Option<Arc<dyn VectorIndex>> {
+        if self.vector_cache.contains_key(key) {
```

Comment:
Should you also record hits/misses from `get_scalar`?

Reply:
Will do!